### PR TITLE
Update o-date documentation

### PR DESCRIPTION
**Pull request by notlee**
- move markup documentation out of the js api readme
- remove docs for apis which now live in `Financial-Times/ft-date-format`
but are passed through for compatibility
- add some JSDoc to remaining apis
- remove CommonJS from demos

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-date bundle size difference from 4.0.7 to c0121d9dce37f6cf56648b899d5fdc7f21edef1e
No significant bundle size differences found.

File: README.md [hunk 1]
```diff
@@ -11,65 +11,79 @@ o-date provides javascript utilities for formatting and updating dates in FT sty
...
+const myDateElement = document.querySelector('#my-date');
+ODate.init(myDateElement);
+```
 
-- `el`: An `HTMLElement` within which to scan for `o-date` elements. If the element itself is a `<time>` element with `o-date` in `data-o-component`, then o-date will run directly on this element rather than querying for suitable elements within it.
+Or dispatch the `o.DOMContentLoaded` event to auto-construct all Origami components on the page, including each `o-date` element with a `data-o-component="o-date"` attribute:
 
-If the `el` is a valid `<time>` element, the resulting o-date instance will be returned; otherwise, an array of created instances will be returned.
+```js
+document.addEventListener('DOMContentLoaded', function() {
```
Comment (JakeChampion) [inline README.md:83]: I think this requires importing o-date first to have [o-date's event listener](https://github.com/Financial-Times/o-date/blob/6ba3e47266aa249de985459cb78fc58c2a7e910f/main.js#L145-L147) setup.
```suggestion
import 'o-date';
document.addEventListener('DOMContentLoaded', function() {
```

**Comment by origamiserviceuser** [comment]
o-date bundle size difference from 4.0.7 to ce93623fb9f149ce5b4e2ad7b4bd2edd8b26df76
No significant bundle size differences found.

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v4.0.8 :tada:

**Pull request merged**